### PR TITLE
FIX wrong stock list with multicompany and without stock sharing

### PR DESCRIPTION
**Pull request by hregis**
@eldy this bug affect v14, v15, v16 and develop

### CODE REVIEW

File: htdocs/product/reassort.php [hunk 1]
```diff
@@ -141,7 +141,7 @@
 $sql .= $hookmanager->resPrint;
 $sql .= ' FROM '.MAIN_DB_PREFIX.'product as p';
 $sql .= ' LEFT JOIN '.MAIN_DB_PREFIX.'product_stock as s ON p.rowid = s.fk_product';
-$sql .= ' LEFT JOIN '.MAIN_DB_PREFIX.'entrepot as e ON s.fk_entrepot = e.rowid AND e.entity IN ('.getEntity('entrepot').')';
+$sql .= ' RIGHT JOIN '.MAIN_DB_PREFIX.'entrepot as e ON s.fk_entrepot = e.rowid AND e.entity IN ('.getEntity('stock').')';
```
Comment (eldy) [inline htdocs/product/reassort.php:144]: Why a RIGHT JOIN. We already have a LEFT JOIN so we take all product whateever is link into product_stock, so adding a RIGHT JOIN here will means we also have another master table. We can't have 2 master table in same SQL as this lead to unpredicable results.

Comment (hregis) [reply htdocs/product/reassort.php:144]: @eldy INNER JOIN ? or move `e.entity IN ('.getEntity('stock').')` after the WHERE ?

Comment (hregis) [reply htdocs/product/reassort.php]: @eldy i replace with INNER JOIN

Comment (eldy) [reply htdocs/product/reassort.php]: If you use a inner join, if the product has not stock, the line will not be returned.

Comment (hregis) [reply htdocs/product/reassort.php]: @eldy "reassort" with no stock (entrepot)  it's possible ? :-)

**Pull request closed without merging**